various-receipes/message-queue/test/unit/mq-example-with-stub.test.js

```diff
@@ -0,0 +1,95 @@
+const request = require("supertest");
+const sinon = require("sinon");
+const { initializeWebServer, stopWebServer } = require("../../../../example-application/api-under-test");
+const { PubSub, Subscription } = require("@google-cloud/pubsub");
+
+let expressApp;
+let sinonSandbox;
+
+beforeAll(async (done) => {
+    // ️️️✅ Best Practice: Place the backend under test within the same process
+    expressApp = await initializeWebServer();
+
+    // ️️️✅ Best Practice: use a sandbox for test doubles for proper clean-up between tests
+    sinonSandbox = sinon.createSandbox();
+
+    done();
+});
+
+afterAll(async (done) => {
+    // ️️️✅ Best Practice: Clean-up resources after each run
+    await stopWebServer();
+    done();
+});
+
+beforeEach(() => {
+    jest.setTimeout(30000);
+    if (sinonSandbox) {
+        sinonSandbox.restore();
+    }
+});
+
+// ️️️✅ Best Practice: Structure tests
+describe("pubsub", () => {
+    describe("/api", () => {
+        describe("POST /emit", () => {
+            test("When emitting a message using pubsub message queue, then should receive message", async () => {
```

Comment:
This sounds like a test of the wrapper, not of a business feature. Is this indeed the goal?